pkgs/applications/audio/soundux/default.nix

```diff
@@ -0,0 +1,136 @@
+{ lib
+, stdenv
+, fetchFromGitHub
+, cmake
+, pkg-config
+, makeBinaryWrapper
+, autoPatchelfHook
+, pipewire
+, libpulseaudio
+, libappindicator
+, libstartup_notification
+, openssl
+, libwnck
+, pcre
+, util-linux
+, libselinux
+, libsepol
+, libthai
+, libdatrie
+, xorg
+, libxkbcommon
+, libepoxy
+, dbus
+, at-spi2-core
+, gnome2
+, webkitgtk
+, yt-dlp
+, ffmpeg
+, lsb-release
+}:
+
+stdenv.mkDerivation rec {
+  pname = "soundux";
+  version = "0.2.7";
+
+  src = fetchFromGitHub {
+    owner = "Soundux";
+    repo = "Soundux";
+    rev = version;
+    sha256 = "sha256-aSCsg6nJt6F+6O7UeXnvYva0vllTfsxK/cjaeOhObZY=";
+    fetchSubmodules = true;
+  };
+
+  nativeBuildInputs = [
+    cmake
+    pkg-config
+    makeBinaryWrapper
+    autoPatchelfHook
+  ];
+
+  buildInputs = [
+    pipewire
+    libpulseaudio
+    libappindicator
+    openssl
+    libwnck
+    pcre
+    util-linux
+    libselinux
+    libsepol
+    libthai
+    libdatrie
+    xorg.libXdmcp
+    xorg.libXtst
+    xorg.libXres
+    libxkbcommon
+    libepoxy
+    dbus
+    at-spi2-core
+    libstartup_notification
+    webkitgtk
+    yt-dlp
+    ffmpeg
+  ];
+
+  patchPhase = ''
+    # Required to make the installation paths relative to the nix output
+    # directory, because soundux's CMakeLists file sets the install prefix to
+    # /opt by force, and hard-codes /usr/share/ paths to some install command
+    # invocations.
+
+    substituteInPlace CMakeLists.txt \
+      --replace "set(CMAKE_INSTALL_PREFIX \"/opt/soundux\" CACHE PATH \"Install path prefix, prepended onto install directories.\" FORCE)" "" \
+      --replace "/usr/share" "$out/usr/share"
+
+    # Soundux hard-codes icon paths.
+    substituteInPlace src/ui/impl/webview/webview.cpp \
+      --replace "/usr/share/pixmaps/soundux.png" "$out/share/pixmaps/soundux.png"
+  '';
+
+  # We need to append /opt to our CMAKE_INSTALL_PREFIX, so we tell Nix not to
+  # add the argument for us.
+  dontAddPrefix = true;
+
+  preConfigure = ''
+    # This needs to be set in preConfigure instead of cmakeFlags in order to
+    # access the $prefix environment variable.
+    export cmakeFlags="-DCMAKE_INSTALL_PREFIX=$prefix/opt $cmakeFlags"
+  '';
+
+  # Somehow some of the install destination paths in the build system still
+  # gets transformed to point to /var/empty/share, even though they are at least
+  # relative to the nix output directory with our earlier patching.
+  postInstall = ''
+    mv "$out/var/empty/share" "$out"
+    rmdir --ignore-fail-on-non-empty --parents "$out/var/empty"
+    mkdir "$out/bin"
+    ln -s "$out/opt/soundux" "$out/bin"
+    substituteInPlace "$out/share/applications/soundux.desktop" \
+      --replace "/opt/soundux/soundux" "soundux"
+  '';
+
+  postFixup = ''
+    patchelf \
+      --add-needed libwnck-3.so \
+      --add-needed libpipewire-0.3.so \
+      --add-needed libpulse.so \
+      "$out/bin/soundux"
```

Comment:
Do we really need to do this manually?

Reply:
GitHub's UI is not making it clear which lines you are commenting on here, so I'll assume you are commenting on both the `postInstall` and `postFixup` phases.

Regarding the `postInstall`, upstream's build system is poorly written, and insists on installing to both `/opt` and `/usr/share` (although the latter gets transformed to `/var/empty/share` during the configure phase somehow, and I have been unable to figure out why).

Regarding the `postFixup`, the `soundux` binary loads some of its dependencies with `dlsym`, so `autoPatchelf` will not add them to the `RPATH` without this here.

Reply:
> GitHub's UI is not making it clear which lines you are commenting on here

Those lines:

![image](https://user-images.githubusercontent.com/7258858/184663801-a5a9fff4-15c1-40b8-91bb-20060cd11a33.png)

Reply:
I have changed this to add the `/lib` paths for these packages to the `RPATH` of the (actual) soundux executable, since they are `dlopen`-ed by Soundux.